src/alexandra_ai_data/domsdatabasen.py

```diff
@@ -0,0 +1,90 @@
+"""API for accessing processed data from Domsdatabasen."""
+
+from omegaconf import DictConfig
+
+from domsdatabasen import DatasetBuilder, Scraper, Processor
+
+from .config_domsdatabasen import config
+
+from logging import getLogger
+
+from datasets import load_dataset
+
+logger = getLogger(__name__)
+
+
+class Domsdatabasen:
+    """API for accessing processed data from Domsdatabasen
+
+    Attributes:
+        config (DictConfig):
+            Configuration settings object.
+        scraper (Scraper):
+            Scraper object for scraping data from Domsdatabasen.
+        processor (Processor):
+            Processor object for processing scraped data.
+        dataset_builder (DatasetBuilder):
+            DatasetBuilder object for building dataset samples.
+        dataset (Dataset):
+            Dataset of processed data from Domsdatabasen.
+    """
+
+    def __init__(self):
+        """Initialize."""
+        self.config: DictConfig = config
+        self.dataset = load_dataset("alexandrainst/domsdatabasen")["train"]
+
+        # The following objects will not be initialized until
+        # the first time they are needed.
+        self.scraper = None
+        self.processor = None
+        self.dataset_builder = None
+
+    def get_case(self, case_id: str) -> dict:
+        """Get processed data for a case from Domsdatabasen.
+
+        If the case_id is already in the dataset, the data is returned from the dataset.
+        Else, the case will be scraped and processed.
+
+        Args:
+            case_id (str):
+                The case_id of the case to get data for.
+
+        Returns:
+            dataset_sample (dict):
+                Processed data for the case.
+        """
+        if isinstance(case_id, int):
+            case_id = str(case_id)
```

Comment:
If integer inputs are allowed, then this should be stated in the type signature.